internal/transport/http2_server.go

```diff
@@ -361,15 +363,12 @@ func (t *http2Server) operateHeaders(frame *http2.MetaHeadersFrame, handle func(
 			rstCode:  http2.ErrCodeFrameSize,
 			onWrite:  func() {},
 		})
-		return false
+		return nil
 	}
 
 	if streamID%2 != 1 || streamID <= t.maxStreamID {
 		// illegal gRPC stream id.
-		if logger.V(logLevel) {
-			logger.Errorf("transport: http2Server.HandleStreams received an illegal stream id: %v", streamID)
-		}
-		return true
+		return fmt.Errorf("transport: http2Server.HandleStreams received an illegal stream id: %v", streamID)
```

Comment:
Per Easwar's comment, I changed this error message to fmt.Errorf("received an illegal stream id: %v", streamID) (the transport: http2Server.HandleStreams part made sense for the logging call but not error). I think it makes more sense to scope this error to only the stream id, as that is what is causing the problem/transport closure. Let me know what you think.

Reply:
I think calling out the stream ID as the problem is good, but then showing the _entire_ frame could be helpful in case there's any other data in there that might be able to help debugging, e.g. the method.

Reply:
ping